at_lookup/lib/src/connection/outbound_message_listener.dart

```diff
@@ -8,50 +8,76 @@ import 'package:at_utils/at_logger.dart';
 ///Listener class for messages received by [RemoteSecondary]
 class OutboundMessageListener {
   final logger = AtSignLogger('OutboundMessageListener');
-  final _buffer = ByteBuffer(capacity: 10240000);
-  late Queue _queue;
+  late ByteBuffer _buffer;
+  final Queue _queue = Queue();
   final _connection;
   Function? syncCallback;
+  final int terminatingChar = 10;
 
-  OutboundMessageListener(this._connection);
+  OutboundMessageListener(this._connection, {int bufferCapacity = 10240000}) {
+    _buffer = ByteBuffer(capacity: bufferCapacity);
+  }
 
   /// Listens to the underlying connection's socket if the connection is created.
   /// @throws [AtConnectException] if the connection is not yet created
   void listen() {
-    _connection.getSocket().listen(_messageHandler,
+    _connection.getSocket().listen(messageHandler,
         onDone: _finishedHandler, onError: _errorHandler);
-    _queue = Queue();
   }
 
   /// Handles messages on the inbound client's connection and calls the verb executor
   /// Closes the inbound connection in case of any error.
   /// Throw a [BufferOverFlowException] if buffer is unable to hold incoming data
-  Future<void> _messageHandler(data) async {
+  Future<void> messageHandler(data) async {
     String result;
-    if (!_buffer.isOverFlow(data)) {
-      // skip @ prompt
-      if (data.length == 1 && data.first == 64) {
-        return;
-      }
-      //ignore prompt(@ or @<atSign>@) after '\n'
-      // If data contains 10(utf8 character for '\n'), trim the string after the '\n'
-      if (data.contains(10)) {
-        data = data.sublist(0, data.lastIndexOf(10) + 1);
-      }
-      _buffer.append(data);
-    } else {
+    // check buffer overflow
+    _checkBufferOverFlow(data);
+    // Add data to buffer until a new line character is encountered
+    _buffer.append(data);
+    // If data does contains terminating character process the data in buffer
+    if (data.contains(terminatingChar)) {
+      result = utf8.decode(_buffer.getData());
```

Comment:
Nope, as I said before, you can only decode what you know is decodable - i.e. everything up to the newline

See the algorithm I proposed in an earlier review